tests/Validation/ValidationValidatorTest.php

```diff
@@ -3880,6 +3880,32 @@ public function testPassingSlashVulnerability()
         $this->assertTrue($v->fails());
     }
 
+    public function testPlaceholdersAreReplaced()
```

Comment:
This test does not actually covers the problem described in the related issue: it does not check whether validated data array contains the correct key or not.

Reply:
Feel free to open a PR with more tests. That's always a good thing :)

Reply:
Already done: #33932